Google.Api.Gax/GaxServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿/*
+ * Copyright 2022 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Google.Api.Gax;
+using System;
+using System.Reflection;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods enabling dependency injection of Google API clients.
+    /// </summary>
+    public static class GaxServiceCollectionExtensions
```

Comment:
That's definitely a possibility. I think it might end up being weird for APIs with a *lot* of clients, e.g. Compute. Wouldn't be hard to try though.

We need to work out whether we definitely want the "Google" part in there. It does make it easier to find...

Reply:
Yep, we will end with lots of extension methods, but that's really not a huge deal, if users are using the generic approach then they'd have to register all the clients that they are using individually as well, one by one.

I think the Google part is good, maybe we can use the first part of the library namespace in the generator instead of the literal Google. It will give us Google for all our libraries and still be general enough for others.